packages/mask/src/plugins/IdeaMarket/SNSAdaptor/ListingsView.tsx

```diff
@@ -0,0 +1,234 @@
+            '& .MuiDataGrid-row:nth-child(even)': {
+                '&:hover': {
+                    backgroundColor: 'inherit',
+                },
+            },
+        },
+        box: {
+            padding: 0.5,
+            paddingBottom: 0,
+        },
+        market: {
+            color: 'rgba(8,87,224,1)',
+            marginLeft: theme.spacing(0.25),
+        },
+        empty: {
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            justifyContent: 'center',
+            height: '100%',
+            padding: theme.spacing(8, 0),
+        },
+        textField: {
+            margin: theme.spacing(1, 0.5, 1.5),
+            '& .MuiSvgIcon-root': {
+                marginRight: theme.spacing(0.5),
+            },
+            '& .MuiInput-underline:before': {
+                borderBottom: 1,
+                borderColor: 'divider',
+            },
+        },
+    }
+})
+
+interface QuickSearchToolbarProps {
+    clearSearch: () => void
+    onChange: () => void
+    value: string
+}
+
+function QuickSearchToolbar(props: QuickSearchToolbarProps) {
+    const { classes } = useStyles()
+
+    return (
+        <Box className={classes.box}>
+            <TextField
+                className={classes.textField}
+                variant="standard"
+                value={props.value}
+                onChange={props.onChange}
+                placeholder="Search&#x2026;"
+                InputProps={{
+                    startAdornment: <SearchIcon fontSize="small" />,
+                    endAdornment: (
+                        <IconButton
+                            title="Clear"
+                            aria-label="Clear"
+                            size="small"
+                            style={{ visibility: props.value ? 'visible' : 'hidden' }}
+                            onClick={props.clearSearch}>
+                            <ClearIcon fontSize="small" />
+                        </IconButton>
+                    ),
+                }}
+            />
+        </Box>
+    )
+}
+
+export function ListingsView() {
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const [searchText, setSearchText] = useState('')
+    const { tokens, loading } = useFetchIdeaTokensBySearch(searchText)
+
+    const formattedData = tokens?.map((token: IdeaToken) => {
+        return {
+            id: token.id,
+            name: token.name,
+            dayChange: token.dayChange,
+            market: token.market.name,
+            price: token.latestPricePoint.price,
+            deposits: formatWeiToEther(token.daiInToken).toNumber(),
+            button: '',
+        }
+    })
+
+    const renderNameCell = (params: GridRenderCellParams<String>) => (
+        <Grid container direction="column">
+            <div title={params.row.name}>{truncate(params.row.name, 25)}</div>
+            <div className={classes.market}>{params.row.market}</div>
```

Comment:
Recommend to use `<Typography />` for textures.

Reply:
Done.